doc/reference/tokenratelimitpolicy.md

```diff
@@ -0,0 +1,539 @@
+kind: TokenRateLimitPolicy
+metadata:
+  name: organisation-overrides
+  namespace: gateway-system
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: Gateway
+    name: api-gateway
+  overrides:
+    strategy: atomic
+    limits:
+      org-wide-limit:
+        rates:
+        - limit: 1000000    # 1M tokens per day org-wide limit
+          window: 1d
+        counter: 'request.auth.claims["org_id"]'
+      security-limit:
+        rates:
+        - limit: 100        # 100 tokens per minute security limit
+          window: 1m
+        predicate: 'request.headers["x-suspicious"] != ""'
+        counter: 'request.headers["x-client-ip"]'
+```
+
+This policy enforces organisation-wide limits that **cannot be overridden** by any HTTPRoute-level policies.
+
+### Policy Hierarchy in Action
+
+```yaml
+# Gateway defaults
+apiVersion: kuadrant.io/v1alpha1
+kind: TokenRateLimitPolicy
+metadata:
+  name: gateway-baseline
+  namespace: gateway-system
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: Gateway
+    name: ai-gateway
+  defaults:
+    limits:
+      default-limit:
+        rates:
+        - limit: 50000      # 50k tokens/day default
+          window: 1d
+        counter: auth.identity.userid
+
+---
+# HTTPRoute specific policy (overrides the gateway default)
+apiVersion: kuadrant.io/v1alpha1
+kind: TokenRateLimitPolicy
+metadata:
+  name: premium-api-limits
+  namespace: api-namespace
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: premium-api-route
+  limits:
+    premium:
+      rates:
+      - limit: 500000      # 500k tokens/day for premium API
+        window: 1d
+      counter: auth.identity.userid
+```
+
+In this scenario:
+- Routes without specific policies get the 50k tokens/day limit from the Gateway defaults
+- The `premium-api-route` gets the 500k tokens/day limit from its specific policy
+
+## Token Usage Tracking
+
+TokenRateLimitPolicy automatically tracks token consumption from AI/LLM responses by monitoring the `usage.total_tokens` field in response bodies. This enables accurate usage-based rate limiting where:
+
+- **Request Phase**: The policy evaluates predicates and descriptors during the request
+- **Response Phase**: The policy extracts actual token usage from the response body
+- **Rate Limiting**: Limitador receives the actual token count as `hits_addend` for precise accounting
+
+### Supported Response Formats
+
+The policy automatically parses token usage from response bodies in the following format:
+```json
+{
+  "usage": {
+    "total_tokens": 150,
+    "prompt_tokens": 100,
+    "completion_tokens": 50
+  }
+}
+```
+
+This is compatible with OpenAI-style API responses and similar AI/LLM services.
+
+## CEL Expressions
+
+TokenRateLimitPolicy supports CEL (Common Expression Language) for both predicates and counters, providing powerful flexibility for complex rate limiting scenarios.
+
+### CEL Context and Available Attributes
```

Comment:
Link to https://github.com/Kuadrant/architecture/blob/main/rfcs/0002-well-known-attributes.md ??